### PR TITLE
fix: Predict empty search screen items

### LINKED ISSUE #24401: [Bug]: Search is inconsistant between Prediction trending vs token trending. Predictions shows a blank UI
### Describe the bug

Category: Trending [x] list
Build number: v7.62.0 (3402)
Device/OS: iPhone 13 Max Pro
Description: Search is consistent between Prediction trending vs token trending. Predictions shows a blank UI

### Expected behavior

Prediction Search matches the search pattern of trending tokens and perps where the list is still visible while the user performs the search

### Screenshots/Recordings

https://github.com/user-attachments/assets/bc5e7459-bd52-43cf-89c5-ac2fafc02f11

### Steps to reproduce

See recording

### Error messages or log output

```shell

```

### Where was this bug found?

Internal release testing

### Version

7.62.0

### Build number

3402

### Build type

None

### Device

Iphone 13

### Operating system

iOS

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

Remove conditional rendering in `PredictSearchOverlay` to always display trending results when the search query is empty.

Previously, the search overlay would show a blank screen when opened without an active search query. This change ensures that the default trending markets are visible, aligning with the behavior of other search sections.

## **Changelog**

CHANGELOG entry: fix: Predict empty search screen items

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-mobile/issues/24401 https://consensyssoftware.atlassian.net/browse/ASSETS-2264

## **Manual testing steps**

1. Go to predict
2. Click search
3. EXPECTED - see some initial predict items even when the search query is empty.

### **After**

https://www.loom.com/share/953e045a02a84103afde9191efedc796

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots. 

---
<a href="https://cursor.com/background-agent?bcId=bc-fdbffcea-e1fc-4117-9d2b-3adfcd5a1b1e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-cursor-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-cursor-light.svg">Open in Cursor</picture></a>&nbsp;<a href="https://cursor.com/agents?id=bc-fdbffcea-e1fc-4117-9d2b-3adfcd5a1b1e"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/open-in-web-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/open-in-web-light.svg">Open in Web</picture></a>

---

> [!NOTE]
> Ensures the Predict search overlay displays default trending markets when no query is entered.
> 
> - In `PredictFeed.tsx`, removes conditional gating around the search results area so it always renders, showing trending results for empty `searchQuery`
> - Updates tests in `PredictFeed.test.tsx` to expect the clear button to disappear after clearing and trending results to be visible with an empty query
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit d51774ff8faecbfb8b50ea6be27bad5cdc64c8b4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: app/components/UI/Predict/views/PredictFeed/PredictFeed.test.tsx [hunk 1]
```diff
@@ -452,7 +452,11 @@ describe('PredictFeed', () => {
       fireEvent.changeText(searchInput, 'test query');
       fireEvent.press(getByTestId('clear-button'));
 
-      expect(queryByTestId('predict-search-result-0')).toBeNull();
+      // After clearing search, the clear button should no longer be visible
+      // (only shows when searchQuery.length > 0)
+      expect(queryByTestId('clear-button')).toBeNull();
```
Comment (Prithpal-Sooriya) [inline app/components/UI/Predict/views/PredictFeed/PredictFeed.test.tsx]: @cursor 

```suggestion
      expect(queryByTestId('clear-button')).not.toBeOnTheScreen();
```

File: app/components/UI/Predict/views/PredictFeed/PredictFeed.tsx [hunk 2]
```diff
@@ -615,36 +615,34 @@ const PredictSearchOverlay: React.FC<PredictSearchOverlayProps> = ({
...
+          <PredictOffline onRetry={refetch} />
+        ) : !marketData || marketData.length === 0 ? (
+          <Box twClassName="flex-1 justify-center items-center p-8">
+            <Text
+              variant={TextVariant.BodyMd}
+              color={TextColor.PrimaryAlternative}
+            >
+              {strings('predict.search_no_markets_found', { q: searchQuery })}
+            </Text>
+          </Box>
```
[the remark this answers is not in the document]
Comment (Prithpal-Sooriya) [reply app/components/UI/Predict/views/PredictFeed/PredictFeed.tsx:635]: Very rare edge case, since empty should be cached and show trending.

File: app/components/UI/Predict/views/PredictFeed/PredictFeed.test.tsx [hunk 3]
```diff
@@ -452,7 +452,11 @@ describe('PredictFeed', () => {
       fireEvent.changeText(searchInput, 'test query');
       fireEvent.press(getByTestId('clear-button'));
 
-      expect(queryByTestId('predict-search-result-0')).toBeNull();
+      // After clearing search, the clear button should no longer be visible
+      // (only shows when searchQuery.length > 0)
+      expect(queryByTestId('clear-button')).not.toBeOnTheScreen();
+      // Trending results should still be visible (new behavior - show trending when search is empty)
```
Comment (Prithpal-Sooriya) [inline app/components/UI/Predict/views/PredictFeed/PredictFeed.test.tsx]: @cusor clean up comment

```suggestion
      // Trending results visible when no search query is empty
```

**Review by caieu — APPROVED** [review]
LGTM
